### PR TITLE
resolve #40 use ioutils.TempDir() for wrapper script

### LINKED ISSUE #40: will not run on systems where /tmp is mounted with noexec
Allowing scripts to be executed from `/tmp` is a security concern for may places (although I don't fully understand why).

In any case, I'm experiencing this problem at present, and it seems the plugin should adhere to environment configuration for temporary files.

### PR DESCRIPTION
**Pull request by chrisalbright**

### CODE REVIEW

File: scripts.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ fi
 # Run the git command
 %v "$@"
 
-`, shell, shell, gitBinary))
+`, shell, os.TempDir(), shell, os.TempDir(), os.TempDir(), os.TempDir(), gitBinary))
```
Comment (abiosoft) [inline scripts.go]: since same value is repeated multiple times, how about using `{temp_dir}` as placeholder and using `strings.Replacer` ?

Comment (chrisalbright) [inline scripts.go]: Sounds good. I'll update the PR.

**Comment by abiosoft** [comment]
Awesome. Thanks.

**Comment by abiosoft** [comment]
Actually, this is what I am referring to https://github.com/abiosoft/caddy-git/blob/master/scripts.go#L122-L129. You do not need to write your own replacer.

**Comment by chrisalbright** [comment]
ah - thanks for the info - I'm not very experienced with go, so I don't know the libraries very well.

**Pull request merged**